worker/internal/benthos/transformers/transform_character_substitution.go

```diff
@@ -48,52 +49,57 @@ Substituted: Ifmmp Xpsme 234@%^
 Note that this does not work for hex values: 0x00 -> 0x1F
 */
 
-func TransformCharacterSubstitution(value string) (*string, error) {
+func TransformCharacterScramble(value string) (*string, error) {
 
-	transformedString := strings.Map(substituteChar, value)
+	transformedString := strings.Map(ScrambleChar, value)
 
 	return &transformedString, nil
 }
 
-func substituteChar(r rune) rune {
-	letterMap := map[rune]rune{
-		'a': 'b', 'b': 'c', 'c': 'd', 'd': 'e', 'e': 'f', 'f': 'g', 'g': 'h', 'h': 'i', 'i': 'j',
-		'j': 'k', 'k': 'l', 'l': 'm', 'm': 'n', 'n': 'o', 'o': 'p', 'p': 'q', 'q': 'r', 'r': 's',
-		's': 't', 't': 'u', 'u': 'v', 'v': 'w', 'w': 'x', 'x': 'y', 'y': 'z', 'z': 'a',
-	}
+func ScrambleChar(r rune) rune {
+	letterList := "abcdefghijklmnopqrstuvwxyz"
 
-	numberMap := map[rune]rune{
-		'0': '1', '1': '2', '2': '3', '3': '4', '4': '5', '5': '6', '6': '7', '7': '8', '8': '9', '9': '0',
-	}
+	numberList := "0123456789"
+
+	specialCharList := "!@#$%^&*()-+=_ []{}|\\ ;\"<>,./?"
 
-	specialCharMap := map[rune]rune{
-		'!': '@', '@': '#', '#': '$', '$': '%', '%': '^', '^': '&', '&': '*',
-		'*': '(', '(': ')', ')': '-', '-': '+', '+': '=', '=': '_',
-		'_': '[', '[': ']', ']': '{', '{': '}', '}': '|', '|': '\\',
-		'\\': ':', ':': ';', ';': '\'', '\'': '"', '"': '<', '<': '>',
-		'>': ',', ',': '.', '.': '/', '/': '?', '?': '!',
+	randStringListInd, err := transformer_utils.GenerateRandomInt64InValueRange(0, int64(len(letterList))-1)
```

Comment:
This is def a micro optimization, but because we know the lists are a constant, we could pre-compute the length and store that as a module-level constant as well, so every function invocation doesn't have to re-calculate the length of the letterlist (or any of the other lists, where as you do right now because it's a string.